hazelcast/src/test/java/com/hazelcast/test/TestLoggerFactory.java

```diff
@@ -33,15 +38,42 @@
  * The real filtering is happening in the log4js once again.
  * Thus it covers branches guarded by is-level-enabled checks
  * yet the real logging is configurable via log4j2.xml
+ *
+ * It also supports changing logging configuration on-the-fly, see {@link TestLoggerFactory#changeConfigFile}
+ * and
  */
-public class TestLoggerFactory implements LoggerFactory {
+public class TestLoggerFactory extends LoggerFactorySupport {
+
+    /**
+     * Log4j XML configuration being currently used. Null indicated the default Log4j behavior.
+     */
+    private static URI configFile = null;
+
+    /**
+     * Store all the logging context being created, because we need to clear them in order to change the configuration
+     * to reload it.
+     */
+    private static final List<LoggerContext> loggerContexts = new ArrayList<LoggerContext>();
```

Comment:
Looks like this field is also may be non-static, since all logger contexts tracked in it should be obtained through the same shared instance of `TestLoggerFactory`.